python/mxnet/rnn/rnn_cell.py

```diff
@@ -0,0 +1,172 @@
+        return (0, self._num_hidden)
+
+    def __call__(self, inputs, states, params, prefix=''):
+        W = params.get('%si2h_weight'%prefix)
+        B = params.get('%si2h_bias'%prefix)
+        U = params.get('%sh2h_weight'%prefix)
+        name = '%st%d_'%(prefix, self._counter)
+        i2h = symbol.FullyConnected(data=inputs, weight=W, bias=B,
+                                    num_hidden=self._num_hidden,
+                                    name='%si2h'%name)
+        h2h = symbol.FullyConnected(data=states, weight=U, no_bias=True,
+                                    num_hidden=self._num_hidden,
+                                    name='%sh2h'%name)
+        output = self._get_activation(i2h + h2h, self._activation,
+                                      name='%sout'%name)
+
+        self._counter += 1
+        return output, output
+
+class LSTMCell(BaseRNNCell):
+    """LSTM cell"""
+    def __init__(self, num_hidden):
+        self._num_hidden = num_hidden
+        self._counter = 0
+
+    @property
+    def state_shape(self):
+        return [(0, self._num_hidden), (0, self._num_hidden)]
+
+    @property
+    def output_shape(self):
+        return (0, self._num_hidden)
+
+    def __call__(self, inputs, states, params, prefix=''):
+        iW = params.get('%si2h_weight'%prefix)
+        iB = params.get('%si2h_bias'%prefix)
+        hW = params.get('%sh2h_weight'%prefix)
+        name = '%st%d_'%(prefix, self._counter)
+        i2h = symbol.FullyConnected(data=inputs, weight=iW, bias=iB,
+                                    num_hidden=self._num_hidden*4,
+                                    name='%si2h'%name)
+        h2h = symbol.FullyConnected(data=states[0], weight=hW, no_bias=True,
+                                    num_hidden=self._num_hidden*4,
+                                    name='%sh2h'%name)
+        gates = i2h + h2h
+        slice_gates = symbol.SliceChannel(gates, num_outputs=4,
+                                          name="%sslice"%name)
+        in_gate = symbol.Activation(slice_gates[0], act_type="sigmoid",
+                                    name='%si'%name)
+        in_transform = symbol.Activation(slice_gates[1], act_type="tanh",
+                                         name='%sc'%name)
+        forget_gate = symbol.Activation(slice_gates[2], act_type="sigmoid",
+                                        name='%sf'%name)
+        out_gate = symbol.Activation(slice_gates[3], act_type="sigmoid",
+                                     name='%so'%name)
+        next_c = symbol._internal._plus(forget_gate * states[1], in_gate * in_transform,
+                                        name='%sstate'%name)
+        next_h = symbol._internal._mul(out_gate, symbol.Activation(next_c, act_type="tanh"),
+                                        name='%sout'%name)
+
+        self._counter += 1
+        return next_h, [next_h, next_c]
+
+
+class StackedRNNCell(BaseRNNCell):
+    """Stacked multple rnn cels"""
```

Comment:
multple -> multiple

This takes a single RNN cell and returns a simple stack of it?  No.  This takes a list of RNN cells and puts them together to make a stack by wiring the outputs to the inputs.  We should say that.

Also, that seems like a bunch of work for somebody to have to do to use a stacked LSTM.  They should just be able to do this more easily, like with a single constructor.  This class seems more like an "RNNCellStacker" than a "StackedRNNCell".